src/main/clm_driver.F90

```diff
@@ -423,11 +425,25 @@ subroutine clm_drv(doalb, nextsw_cday, declinp1, declin, rstwr, nlend, rdate, ro
        call setExposedvegpFilter(bounds_clump, &
             canopystate_inst%frac_veg_nosno_patch(bounds_clump%begp:bounds_clump%endp))
 
+       ! Amount of water available for groundwater irrigation
+       call CalcAvailableUnconfinedAquifer(bounds_clump, filter(nc)%num_hydrologyc, &
+            filter(nc)%hydrologyc, soilhydrology_inst, soilstate_inst, &
+              water_inst%waterdiagnosticbulk_inst)
+
        ! Irrigation flux
        ! input is main channel storage
-       call irrigation_inst%ApplyIrrigation(bounds_clump, water_inst%waterfluxbulk_inst)
+       call irrigation_inst%ApplyIrrigation(bounds_clump, filter(nc)%num_soilc, &
+            filter(nc)%soilc, filter(nc)%num_soilp, filter(nc)%soilp, &
+            water_inst%waterfluxbulk_inst, &
+            available_gw_uncon = water_inst%waterdiagnosticbulk_inst%available_gw_uncon_col(bounds_clump%begc:bounds_clump%endc))
        call t_stopf('drvinit')
 
+       ! Remove groundwater irrigation
+       call WithdrawGroundwaterIrrigation(bounds_clump, filter(nc)%num_hydrologyc, &
```

Comment:
Were you going to wrap this in a conditional on whether we're using groundwater irrigation? I don't care strongly if that's done, though I did like your suggestion of doing it. If so, would it be correct to also wrap CalcAvailableUnconfinedAquifer in a similar conditional?

Since the irrigation params are private, we'd need to put in a simple logical function in IrrigationMod that returns the value of this%params%use_groundwater_irrigation:

```Fortran
  !-----------------------------------------------------------------------
  function UseGroundwaterIrrigation(this)
    !
    ! !DESCRIPTION:
    ! Returns true if we're using groundwater irrigation in this urn
    !
    ! !ARGUMENTS:
    logical :: UseGroundwaterIrrigation  ! function result
    class(irrigation_type), intent(in) :: this
    !-----------------------------------------------------------------------

    UseGroundwaterIrrigation = this%params%use_groundwater_irrigation

  end function UseGroundwaterIrrigation
```

Reply:
@swensosc wrapped the call to WithdrawGroundwaterIrrigation in a conditional. He said he could go either way on the call to CalcAvailableUnconfinedAquifer... I might add a conditional around that in a final cleanup.